src/parquet/arrow/writer.cc

```diff
@@ -679,7 +678,7 @@ Status FileWriter::Impl::TypedWriteBatch<BooleanType, ::arrow::BooleanType>(
     const int16_t* def_levels, const int16_t* rep_levels) {
   RETURN_NOT_OK(data_buffer_.Resize(array->length()));
   auto data = static_cast<const BooleanArray*>(array.get());
-  auto data_ptr = reinterpret_cast<const uint8_t*>(data->values()->data());
+  auto data_ptr = data->values()->data();
```

Comment:
Google styleguide nitpick: Only use `auto` where the type of the variable is already mentioned in the line.